movieapp/src/containers/SortElement/SortElement.tsx

```diff
@@ -1,29 +1,43 @@
 import { SortElement } from './SortElementTypes';
 import { Option } from '../../components/Option/Option';
+import { useDispatch } from 'react-redux';
+import { receiveMoviesFromApiError, sortMoviesById } from '../../actions/actions';
+import { AnyAction } from 'redux';
 
 const movieSortCriterias: Array<SortElement> = [
   {
     id: '1',
     title: 'Popularity',
-    link: 'popularity'
+    value: 'vote_count'
   },
   {
     id: '2',
     title: 'Release date',
-    link: 'release-date'
+    value: 'release_date'
   },
   {
     id: '3',
     title: 'Rating',
-    link: 'rating'
+    value: 'vote_average'
   }
 ];
 
 export function SortElement(): JSX.Element {
+  const dispatch = useDispatch();
+
+  function SortMovies(e: any): void {
+    fetch('http://localhost:4000/movies?sortBy=' + e.currentTarget.value + '&sortOrder=asc')
+      .then((res) => res.json())
+      .then((movies) => {
+        dispatch(sortMoviesById(movies.data) as AnyAction);
+      })
+      .catch((error) => dispatch(receiveMoviesFromApiError(error) as AnyAction));
+  }
+
   return (
-    <select>
+    <select onChange={SortMovies}>
```

Comment:
handlers should be called with lowercase - it's just function (not class or enum

Reply:
fixed